### PR TITLE
Add automatic integration with Conjure

### PR DESCRIPTION
**Pull request by russtoku**
Users won't have to manually configure things to use this plugin with Conjure.

Changes made:
- Automatic integration with Conjure.
- Disable diagnostics in log buffer (not done in the module's code).
- Add sample TypeScript file (for developers).
- Rename module from deno.deno to conjure.client.deno.stdio (to conform to other Conjure client modules).

### CODE REVIEW

**Comment by sigmaSd** [comment]
Thanks again for this work

changing `local autoload = (require("aniseed.autoload")).autoload` to `local autoload = (require("deno.aniseed.autoload")).autoload` indeed makes the error goes away, but now I noticed that I'm seeing diagnostics in the log buffer
image

so to recap
- I'm using lazy.nvim as a plugin manager, any idea how can I use the newest version of aniseed?
- The diagnostics are showing up in the log buffer

**Comment by sigmaSd** [comment]
so aniseed is located under `conjure-deno/lua/deno/aniseed` isn't it expected that the correct require is `require("deno.aniseed.autoload")`

why would `require("aniseed.autoload")` work ? unless you have aniseed installed as a separate package

**Comment by sigmaSd** [comment]
I get it now, the script that aniseed use will probably try to use aniseed from the first location it finds it in, I'm going to assume that you have aniseed installed as a separate package that's why it detected it and used it.

I update aniseed to minimize the diff, please rebase, and edit deno manually

I think this problem is not solved though : The diagnostics are showing up in the log buffer

**Comment by russtoku** [comment]
The *after/plugin/deno.vim* should have create the *conjure_log_buf* group. You can confirm this with `au conjure_log_buf` run in command-line mode.
```
:au conjure_log_buf                                                                                            
--- Autocommands ---                                                                                           
conjure_log_buf  BufNewFile                                                                                    
    'conjure-log-*'                                                                                            
              lua vim.diagnostic.disable(0)
```
Also, you can run `scr` in command-line mode to see all of the scripts that have been loaded. *after/plugin/deno.vim* should be in that list.

I do have Aniseed installed along with Conjure so that may be why `require("aniseed.autoload")` works for me.

**Comment by sigmaSd** [comment]
I do have the group indeed, but for some reason its not triggered

can you rebase, edit deno manually and switch from deno.vim to deno.lua (I prefer lua wherever possible)

I'll debug the diagnostic problem after theses changes

**Comment by russtoku** [comment]
> can you rebase, edit deno manually and switch from deno.vim to deno.lua (I prefer lua wherever possible)

Yes, will do.

**Comment by russtoku** [comment]
Still working on convertin `deno.vim` to `deno.lua`.

File: after/plugin/deno.vim [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+augroup conjure_init_filetypes
+  au FileType typescript  lua require('conjure.mapping')['on-filetype']()
+  au FileType javascript  lua require('conjure.mapping')['on-filetype']()
+augroup END
+
+" Disable diagnostics on the log buffer
+"   - Moved here from fnl/deno/deno.fnl and translated to Vimscript.
+augroup conjure_log_buf
+  au!
+  au BufNewFile 'conjure-log-*' lua vim.diagnostic.disable(0)
```
Comment (sigmaSd) [inline after/plugin/deno.vim]: `'conjure-log-*'`  to  `conjure-log-*` (no quotes) to fix the issue

Comment (russtoku) [reply after/plugin/deno.vim]: Ah...silly me!

**Comment by sigmaSd** [comment]
I added one small commit https://github.com/sigmaSd/conjure-deno/pull/1/commits/d7bab0b7ca44a8c4ebaf8d6917ca3e1206832ccf

thanks again!

**Comment by russtoku** [comment]
Glad this worked out for your plugin.

Thank you, too! I learned some stuff.

**Pull request merged**